src/ProfileEndpoints.jl

```diff
@@ -281,6 +310,9 @@ function __init__()
         precompile(_start_alloc_profile, (Float64,)) || error("precompilation of package functions is not supposed to fail")
         precompile(_stop_alloc_profile, ()) || error("precompilation of package functions is not supposed to fail")
     end
+    
+    precompile(typeinf_start_endpoint, (HTTP.Request,)) || error("precompilation of package functions is not supposed to fail")
+    precompile(typeinf_stop_endpoint, (HTTP.Request,)) || error("precompilation of package functions is not supposed to fail")
```

Comment:
i don't know why we did all of these this way, but it seems a bit silly now...

Can we maybe switch them all to `@assert precompile(...)` instead of `precompile(...) || error(...)`? It's shorter, easier to read, and easier to copy/paste. If you don't want to do that change in this PR tho, we can do it later 👍